chemprop/cli/common.py

```diff
@@ -61,12 +63,10 @@ def add_common_args(parser: ArgumentParser) -> ArgumentParser:
     featurization_args.add_argument(
         "--keep-h",
         action="store_true",
-        help="Whether H are explicitly specified in input (and should be kept this way). This option is intended to be used with the :code:`reaction` or :code:`reaction_solvent` options, and applies only to the reaction part.",
+        help="Whether H's explicitly specified in input should be kept in the mol graph.",
     )
     featurization_args.add_argument(
-        "--add-h",
-        action="store_true",
-        help="Whether RDKit molecules will be constructed with adding the Hs to them. This option is intended to be used with Chemprop's default molecule or multi-molecule encoders, or in :code:`reaction_solvent` mode where it applies to the solvent only.",
+        "--add-h", action="store_true", help="Whether H's should be added to the mol graph."
```

Comment:
What is the difference of having add h and keep h?

Reply:
add_h will add hydrogens to the mol graph at any locations that should have hydrogens. keep_h will keep the hydrogens you specify in the SMILES string. Does that make more sense? I can try explaining it in more detail.

Reply:
Thanks for explaining! Do keep_h imply add_h? I guess I'm confused how they interact with each other.

Reply:
Looking at the source code:
```
    if keep_h:
        mol = Chem.MolFromSmiles(smi, sanitize=False)
        Chem.SanitizeMol(
            mol, sanitizeOps=Chem.SanitizeFlags.SANITIZE_ALL ^ Chem.SanitizeFlags.SANITIZE_ADJUSTHS
        )
    else:
        mol = Chem.MolFromSmiles(smi)

    return Chem.AddHs(mol) if add_h else mol
```
My understanding is that it first checks keep_h and if true doesn't remove the hydrogens that are explicit in the SMILES string. This means those hydrogens keep their atom index based on the original SMILES graph. Then if add_h is true, more hydrogens will be added at all locations that don't have hydrogens and their atom indexes will be after all the original atoms in the graph. 

Disclaimer though that I haven't tested this.